console/src/hardware/ni/task/AnalogRead.tsx

```diff
@@ -199,13 +201,65 @@ const Wrapped = ({
     },
   });
 
+  const copy = useCopyToClipboard();
+  const handleCopyPythonCode = () => {
+    const name = methods.get("name").value;
+    copy(
+      `
+      from synnax.hardware.ni import AnalogReadTask
+      # Retrieve ${name}
+      task = AnalogReadTask(client.hardware.tasks.retrieve(key=${task?.key}))
+      `,
+      `Python code for ${name}`,
+    );
+  };
+
+  const handleCopyAsJSON = () => {
+    const name = methods.get("name").value;
+    copy(
+      binary.JSON_CODEC.encodeString(methods.value().config),
+      `configuration JSON for ${name}`,
+    );
+  };
+
   return (
     <Align.Space className={CSS.B("task-configure")} direction="y" grow empty>
       <Align.Space grow>
-        <Form.Form {...methods}>
-          <Form.Field<string> path="name">
-            {(p) => <Input.Text variant="natural" level="h1" {...p} />}
-          </Form.Field>
+        <Form.Form {...methods} mode={task?.snapshot ? "preview" : "normal"}>
+          <Align.Space direction="x" justify="spaceBetween">
+            <Form.Field<string>
+              path="name"
+              showLabel={false}
+              padHelpText={!task?.snapshot}
+            >
+              {(p) => (
+                <Input.Text
+                  variant={task?.snapshot ? "preview" : "natural"}
+                  level="h1"
+                  {...p}
+                />
+              )}
+            </Form.Field>
+            <Align.Space direction="x" size="small">
+              <Button.Icon
+                tooltip={`Copy Python code for ${methods.get("name").value}`}
+                tooltipLocation="left"
+                variant="text"
+                onClick={handleCopyPythonCode}
+              >
+                <Icon.Python style={{ color: "var(--pluto-gray-l7)" }} />
+              </Button.Icon>
+              <Button.Icon
```

Comment:
Why not add TypeScript code here as well? Also shouldn't these buttons get added to the tasks for OPC UA / other NI tasks as well?

Reply:
We currently don't have support for configuring and running tasks from the TypeScript client. We'll add a typescript copy button when we do. 

Same goes for OPC UA.

I added copy buttons to the other NI tasks.